### PR TITLE
Feature/102

### PR DESCRIPTION
**Pull request by adilsuccessive**
put functionality of user.
[issue/102](https://github.com/nodexpertsdev/RnD/issues/102)

### CODE REVIEW

File: expressEngine/cms/user/error.js [hunk 1]
```diff
@@ -7,4 +7,7 @@ export default {
   loginError: 'login failed please signUp first',
   passwordError: 'incorrect password',
   tokenError: 'token is not generated',
+  unableToUpdate: 'User not found to update',
+  emptyData: 'Please provide data to update',
```
Comment (chiragSuccessive) [inline expressEngine/cms/user/error.js]: @adilsuccessive  emptyData should be placed in Common cms.

File: expressEngine/cms/user/error.js [hunk 2]
```diff
@@ -7,4 +7,7 @@ export default {
   loginError: 'login failed please signUp first',
   passwordError: 'incorrect password',
   tokenError: 'token is not generated',
+  unableToUpdate: 'User not found to update',
```
Comment (chiragSuccessive) [inline expressEngine/cms/user/error.js]: @adilsuccessive please change unableToUpdate, if data to update is already there in the database then nModified will give 0 and unableToUpdate message will be displayed, but this message is related to user not found, you can set generic value 'unable to update'.

**Comment by gkgbunny** [comment]
After updating user with new field, not existing in schema, it is not showing error message, so please fix it.

File: expressEngine/user/service/user.js [hunk 3]
```diff
@@ -88,6 +89,23 @@ class Service {
...
+      return { error: error.notUpdatable };
+    }
+    if (dataToUpdate.email && !modelLib.validateEmail(dataToUpdate.email)) {
+      return { error: `${dataToUpdate.email} is not a valid email!` };
+    }
+    const updated = await DBService.updateOne(User, { userId: id }, dataToUpdate);
+    return (updated.nModified
+      ? { message: success.userUpdated }
+      : { error: error.unableToUpdate });
+  }
```
Comment (chiragSuccessive) [inline expressEngine/user/service/user.js]: @adilsuccessive please handle db errors.

File: expressEngine/user/service/user.js [hunk 4]
```diff
@@ -88,6 +89,23 @@ class Service {
...
+    if (dataToUpdate.password) {
+      return { error: error.notUpdatable };
+    }
+    if (dataToUpdate.email && !modelLib.validateEmail(dataToUpdate.email)) {
+      return { error: `${dataToUpdate.email} is not a valid email!` };
+    }
+    const updated = await DBService.updateOne(User, { userId: id }, dataToUpdate);
+    return (updated.nModified
+      ? { message: success.userUpdated }
+      : { error: error.unableToUpdate });
```
Comment (chiragSuccessive) [inline expressEngine/user/service/user.js]: @adilsuccessive unableToUpdate error is not returning when the user enters the wrong field in dataToUpdate.

File: expressEngine/cms/user/error.js (hunk 1 above)
Comment (adilsuccessive) [reply expressEngine/cms/user/error.js]: Done

File: expressEngine/cms/user/error.js (hunk 2 above)
Comment (adilsuccessive) [reply expressEngine/cms/user/error.js]: User not found to update is displayed if no records found to update during the search.

**Comment by adilsuccessive** [comment]
> After updating user with new field, not existing in schema, it is not showing error message, so please fix it.

@gkgbunny this should handle in the middleware in the parser to accept only predefined field of the schema.

File: expressEngine/user/service/user.js (hunk 3 above)
Comment (adilsuccessive) [reply expressEngine/user/service/user.js]: Done

File: expressEngine/user/service/user.js (hunk 4 above)
Comment (adilsuccessive) [reply expressEngine/user/service/user.js]: @chiragSuccessive this should handle in the middleware in the parser to accept only predefined field of the schema.

**Comment by adilsuccessive** [comment]
> @adilsuccessive Please fix suggested changes

fixed the changes.

**Pull request merged**